tensorflow/compiler/mlir/tensorflow/transforms/promote_resources_to_args.cc

```diff
@@ -304,7 +304,7 @@ LogicalResult PromoteResourcesToArguments(
       continue;
     }
 
-    const auto index = resource_and_index.index();
+    const long int index = resource_and_index.index();
```

Comment:
In this case I would expect `auto` to determine the right type

Reply:
I do this to resolve a [Wsign-compare] that occurs on 308. The type observed to be assigned by `auto` is long int.

Reply:
Setting at least one of the comparison variables necessarily resolves the warning.